src/homedelivery/export.ts

```diff
@@ -182,12 +185,73 @@ async function processSubs(
 	console.log('loaded ' + holidaySuspensions.size + ' holiday suspensions');
 	const csvFormatterStream = csvFormatterForSalesforce(outputHeaders);
 
+	// Validation counters for CloudWatch metrics
+	let totalRowsProcessed = 0;
+	let missingDeliveryAgentCount = 0;
+	let missingAddressCount = 0;
+	let missingCityCount = 0;
+	let missingPostcodeCount = 0;
+	let missingNameCount = 0;
+
 	const writeRowToCsvStream = (
 		row: Partial<InputRow>,
 		csvStream: csv.CsvFormatterStream<csv.FormatterRow, csv.FormatterRow>,
 	) => {
 		const subscriptionName = row[SUBSCRIPTION_NAME];
 		if (!holidaySuspensions.has(subscriptionName || '')) {
+			totalRowsProcessed++;
+
+			// Validate critical fields (incident-driven)
+			if (!row[DELIVERY_AGENT] || row[DELIVERY_AGENT]?.trim() === '') {
+				missingDeliveryAgentCount++;
+				console.warn(
+					`VALIDATION ERROR: Missing delivery agent for subscription ${subscriptionName}`,
+				);
+			}
+
+			if (!row[ADDRESS_1] || row[ADDRESS_1]?.trim() === '') {
+				missingAddressCount++;
+				const customerName = getFullName(
+					row[FIRST_NAME] || '',
+					row[LAST_NAME] || '',
+				);
+				console.warn(
+					`VALIDATION ERROR: Missing address | Subscription: ${subscriptionName} | Customer: ${customerName} | City: ${row[CITY] || 'N/A'} | Postcode: ${row[POSTAL_CODE] || 'N/A'}`,
+				);
+			}
+
+			if (!row[CITY] || row[CITY]?.trim() === '') {
+				missingCityCount++;
+				const customerName = getFullName(
+					row[FIRST_NAME] || '',
+					row[LAST_NAME] || '',
+				);
+				console.warn(
+					`VALIDATION ERROR: Missing city | Subscription: ${subscriptionName} | Customer: ${customerName} | Address: ${row[ADDRESS_1] || 'N/A'} | Postcode: ${row[POSTAL_CODE] || 'N/A'}`,
+				);
+			}
+
+			if (!row[POSTAL_CODE] || row[POSTAL_CODE]?.trim() === '') {
+				missingPostcodeCount++;
+				const customerName = getFullName(
+					row[FIRST_NAME] || '',
+					row[LAST_NAME] || '',
+				);
+				console.warn(
+					`VALIDATION ERROR: Missing postcode | Subscription: ${subscriptionName} | Customer: ${customerName} | Address: ${row[ADDRESS_1] || 'N/A'} | City: ${row[CITY] || 'N/A'}`,
+				);
+			}
+
+			if (
+				(!row[FIRST_NAME] || row[FIRST_NAME]?.trim() === '') &&
+				(!row[LAST_NAME] || row[LAST_NAME]?.trim() === '')
```

Comment:
Only last name is mandatory, first name is actually optional (and we do have a few without first names). So you can just check last name here.